apps/user-profile/src/app/pages/settings-page/settings-mailing-lists/settings-mailing-lists.component.ts

```diff
@@ -90,24 +166,67 @@ export class SettingsMailingListsComponent implements OnInit {
   }
 
   getOptOutAttribute(resource: RichResource): void {
+    this.selectedResource = resource.name;
     this.index = this.resources.indexOf(resource);
     this.optOutAttribute = this.optOuts[this.index].attribute;
   }
 
-  setOptOut(): void {
-    this.optOuts[this.index].attribute.value = this.optOutAttribute.value
-      ? null
-      : ('true' as unknown as object);
+  unsubscribe(): void {
+    this.optOuts[this.index].attribute.value = 'true' as unknown as object;
+    this.attributesManagerService
+      .setMemberResourceAttribute(this.optOuts[this.index])
```

Comment:
Adam wanted to have some reaction if this method fails(this method is also in subscribe method
my proposal would be when this method fails you just need to change the visuals of the checkbox to represent the state that the operation didn't happen
the error notification will be there automatically, you don't need to worry about that